.github/workflows/build.yml

```diff
@@ -0,0 +1,53 @@
+name: Build Project
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        bundler-cache: true # Runs bundle install and caches gems. See the ruby_test.yml
+                            # example if you need more control over bundler.
+
+    - name: Set release tag
+      run: echo "RELEASE_TAG=$([ ${GITHUB_REF##*/} = "develop" ] && echo "${GITHUB_REF##*/}/$(date +'%Y-%m-%d/%H%M%S')" || printf '%s\n' $(cat .release-version))" >> $GITHUB_ENV
+
+    - name: Create Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+      with:
+        tag_name: ${{ env.RELEASE_TAG }}
+        release_name: ${{ env.RELEASE_TAG }}
+        draft: false
+        prerelease: false
+
+    - name: Set up environment
```

Comment:
Hmm, actually, on reflection I wonder if it makes more sense to split this into two actions. One that makes the release, and then another that makes the build. If the latter is triggered on
```
on:
  release:
    types: published
```
Then I *think* one would automatically chain into the other, but you'd leave scope to create a manual release as well.

Reply:
yes that sounds like a good idea ill look into that

Reply:
Haha, I've just gone to pick up quanthub, and it looks like it follows a similar release pattern. So I'm going to be referencing what you've done here a bit. Let you know if I find out anything that might be useful.

Reply:
So used `${GITHUB_REF#refs/heads/}` to extract the branch name. Not hugely significant in this case, where we're just matching develop and master, but it means it will correctly handle branches with / in the name. Mostly made the change in my version just in case someone tries to nab the code for some other purpose.

Reply:
I've also used:
`prerelease: ${{ github.ref == 'refs/heads/devel' }}`
To automatically flag devel releases as pre-releases